packages/kas/src/nodes.ts

```diff
@@ -2817,49 +2890,52 @@ export class Eq extends Expr {
     // Assumptions: Expression is of the form a+bx, and we solve for x
     solveLinearEquationForVariable(variable) {
         var expr = this.asExpr();
-        if (!expr.is(Add) || expr.terms.length !== 2) {
+        if (!(expr instanceof Add) || expr.terms.length !== 2) {
             throw new Error(
                 "Can only handle linear equations of the form " +
                     "a + bx (= 0)",
             );
         }
 
-        var hasVar = function (term) {
+        var hasVar = function (term: Expr) {
             return term.has(Var) && _.contains(term.getVars(), variable.symbol);
         };
 
-        var a, b;
-
-        if (hasVar(expr.terms[0])) {
-            a = Mul.handleNegative(expr.terms[1]);
-            b = Mul.handleDivide(expr.terms[0], variable);
-        } else {
-            a = Mul.handleNegative(expr.terms[0]);
-            b = Mul.handleDivide(expr.terms[1], variable);
-        }
+        const [a, b] = hasVar(expr.terms[0])
+            ? [
+                  Mul.handleNegative(expr.terms[1]),
+                  Mul.handleDivide(expr.terms[0], variable),
+              ]
+            : [
+                  Mul.handleNegative(expr.terms[0]),
+                  Mul.handleDivide(expr.terms[1], variable),
+              ];
 
         return Mul.handleDivide(a, b).simplify();
     }
 }
 
 /* abstract symbol node */
-class Symbol extends Expr {
+class Sym extends Expr {
```

Comment:
I renamed this to `Sym` so that I could use `Symbol` for the prefixes types down below.

Reply:
Even though we could do something other than using `Symbol` for the prefix types, I'd like avoid shadowing built-ins like this.